src/main/java/io/supertokens/inmemorydb/queries/TOTPQueries.java

```diff
@@ -53,27 +76,85 @@ public static void createDevice(Start start, TOTPDevice device)
         });
     }
 
+    public static void createDeviceAndUser(Start start, TOTPDevice device)
+            throws StorageQueryException, StorageTransactionLogicException {
+        start.startTransaction(con -> {
+            Connection sqlCon = (Connection) con.getConnection();
+
+            try {
+                insertUser_Transaction(start, sqlCon, device.userId);
+                insertDevice_Transaction(start, sqlCon, device);
+                sqlCon.commit();
+            } catch (SQLException e) {
+                throw new StorageTransactionLogicException(e);
+            }
+
+            return null;
+        });
+    }
+
     public static int markDeviceAsVerified(Start start, String userId, String deviceName)
```

Comment:
It does care, right? 
```
            int updatedCount = TOTPQueries.markDeviceAsVerified(this, userId, deviceName);
            if (updatedCount == 0) {
                throw new UnknownDeviceException();
            }
```

Reply:
What happens if the device was already verified? In that case will it return 0 rows updated?

Reply:
It returns 1, not 0 even when device is already verified. Added tests and it passed. Checked manually for Postgres. It's also working.. It returns rows matched not rows updated (Actually postgres has MVCC so it will create new rows anyways)

Also, for MySQL it returned:

```
mysql> UPDATE totp_user_devices SET verified = true WHERE user_id = 'user1' AND device_name = 'device1';
Query OK, 1 row affected (0.00 sec)
Rows matched: 1  Changed: 1  Warnings: 0

mysql> UPDATE totp_user_devices SET verified = true WHERE user_id = 'user1' AND device_name = 'device1';
Query OK, 0 rows affected (0.00 sec)
Rows matched: 1  Changed: 0  Warnings: 0
```

So in the 2nd (repeated) attempt, it shows that 0 rows were changed but rows matched is still the same..

Reply:
We decided to change the markDeviceAsVerified in storageLayer to make it void

Reply:
Done.

Reply:
But keep this comment open. Will cross check the return value of `update` in mysql plugin (coz it returns both)

Reply:
It worked with MySQL 8 as well.  

`update` returned matchedCount, not updateCount, which is exactly what we wanted.